probs/probs.go

```diff
@@ -32,6 +32,7 @@ const (
 	UnauthorizedProblem          = ProblemType("unauthorized")
 	UnsupportedContactProblem    = ProblemType("unsupportedContact")
 	UnsupportedIdentifierProblem = ProblemType("unsupportedIdentifier")
+	ConflictProblem              = ProblemType("conflict")
```

Comment:
Can you move this up between `CAAProblem` and `ConnectionProblem` _or_ something like this
```
// Extra error types not included in the original RFC8555 Section 6.7.
ConflictProblem = ProblemType("conflict")
```

Reply:
Also, this new ProblemType appears to not be used. @beautifulentropy you mentioned in a previous chat that [`probs.Conflict()`](https://github.com/letsencrypt/boulder/blob/8d169a8dfb530d8672dad3f568dfe92e53770604/probs/probs.go#L295) already exists, but it uses `MalformedProblem` as the problem type. This PR calls `probs.Conflict()` below, so we wind up with `MalformedProblem`.

The only two callers of `probs.Conflict()` are in the KeyRollover path. Let's change the behavior of `probs.Conflict()` so it uses this new `ConflictProblem` constant. This will be a user-visible change but it's so tiny I don't think we need to worry about it - the conflict cases in that code path are basically never hit.